lib/slack-ruby-bot/mvc/controller/base.rb

```diff
@@ -0,0 +1,111 @@
+module SlackRubyBot
+  module MVC
+    module Controller
+      class Base
+        include ActiveSupport::Callbacks
+
+        class << self
+          attr_reader :abstract
+          alias abstract? abstract
+
+          def controllers
+            Base.instance_variable_get(:@controllers)
```

Comment:
What would it take to avoid all the `instance_variable_*` calls and use proper public methods? What's the drawback?